src/udev/net/ethtool-util.c

```diff
@@ -665,3 +692,56 @@ int ethtool_set_channels(int *fd, const char *ifname, netdev_channels *channels)
 
         return 0;
 }
+
+int config_parse_advertise(const char *unit,
+                           const char *filename,
+                           unsigned line,
+                           const char *section,
+                           unsigned section_line,
+                           const char *lvalue,
+                           int ltype,
+                           const char *rvalue,
+                           void *data,
+                           void *userdata) {
+        link_config *config = data;
+        NetDevAdvertise mode, a = 0;
+        const char *p;
+        int r;
+
+        assert(filename);
```

Comment:
I'm not familiar with all the config_parse_ functions throughout the code base. But shouldn't they all be asserting on unit too?

Reply:
No. For .link files, `unit` is `NULL`. See the following. https://github.com/systemd/systemd/blob/fd790d6f09b10a87b007b71403cb018f18ff91c9/src/udev/net/link-config.c#L161-L164